boards: nrf9160dk: Change default revision to 0.14.0

gregersrygg opened the pull request:
Change the default revision of the nRF9160 DK to 0.14.0 to make it easier to use external flash. The v0.14.0 revision is almost 3 years old.

It will still be possible to build without external flash support for older boards when specifying a revision older than v0.14.0:

```bash
west build -b nrf9160dk_nrf9160_ns@0.7.0
```

gregersrygg committed:
boards: nrf9160dk: Change default revision to 0.14.0

Change the default revision of the nRF9160 DK to 0.14.0 to make it
easier to use external flash.

Signed-off-by: Gregers Gram Rygg <gregers.gram.rygg@nordicsemi.no>

anangl commented:
> Change the default revision of the nRF9160 DK to 0.14.0 to make it easier to use external flash. The v0.14.0 revision is almost 3 years old.

I think that sounds reasonable. But first we need to apply corrections from #58136 so that CI for this PR has a chance to become green.

anangl commented:
@gregersrygg Could you also add a related release note?

gregersrygg committed:
boards: nrf9160dk: Enable external flash switch in board controller

Enable the external_flash_pins_routing switch in the board controller
overlay for nRF9160 DK revision 0.14.0 and newer. This is to match the
behaviour of the factory shipped borad controller on board revisions
that have external flash.

Signed-off-by: Gregers Gram Rygg <gregers.gram.rygg@nordicsemi.no>

anangl commented on doc/releases/release-notes-3.4.rst:
```diff
@@ -305,6 +305,13 @@ Boards & SoC Support
+    older revision of the nrf9160dk without external flash, specify an
+    older board revision when building.
+
+  * Enable external_flash_pins_routing switch in ``nrf9160dk_nrf52840`` by default.
```
```suggestion
  * Enabled external_flash_pins_routing switch in ``nrf9160dk_nrf52840`` by default.
```

anangl commented on doc/releases/release-notes-3.4.rst:
```diff
@@ -305,6 +305,13 @@ Boards & SoC Support
 
+  * Changed the default board revision to 0.14.0 for the Nordic boards
+    ``nrf9160dk_nrf9160`` and ``nrf9160dk_nrf52840``. To build for an
+    older revision of the nrf9160dk without external flash, specify an
+    older board revision when building.
```
```suggestion
    older revision of the nRF9160 DK without external flash, specify that
    older board revision when building.
```
